### PR TITLE
batches: auto-delete Gerrit Change on close

### PR DESCRIPTION
**Pull request by courier-new**
Closes https://github.com/sourcegraph/sourcegraph/issues/52386.

If the `batchChanges.autoDeleteBranch` site configuration option is enabled, Batch Changes will delete a Change on Gerrit after it is closed/abandoned. It is not possible to delete a Change once it has been merged.

https://github.com/sourcegraph/sourcegraph/assets/8942601/5d9f52cc-b34a-4ef1-a74f-2dc72d39b572

## Test plan

Client unit test coverage. Manual integration test, see video above.

### CODE REVIEW

File: enterprise/internal/batches/sources/gerrit.go [hunk 1]
```diff
@@ -168,6 +169,12 @@ func (s GerritSource) CloseChangeset(ctx context.Context, cs *Changeset) error {
 		return errors.Wrap(err, "abandoning change")
 	}
 
+	if conf.Get().BatchChangesAutoDeleteBranch {
```
Comment (varsanojidan) [inline enterprise/internal/batches/sources/gerrit.go:172]: Could we add a scenario for this in `enterprise/internal/batches/sources/gerrit_test.go`?

**Review by varsanojidan — APPROVED** [review]
Thank you for doing this 🙏 !

File: enterprise/internal/batches/sources/gerrit.go (hunk 1 above)
Comment (courier-new) [reply enterprise/internal/batches/sources/gerrit.go:172]: Can do. 😄

**Comment by varsanojidan** [comment]
🔥 🔥

**Pull request merged**